templates/build.and.release.invokebuild.yml

```diff
@@ -0,0 +1,183 @@
+  postCreateGitHubRelease: []
+  prePublishNugetPackages: []
+  postPublishNugetPackages: []
+  vmImage: ''
+  service_connection_nuget_org: '' 
+  service_connection_github: '' 
+  build_script: ''
+  build_task: '.'
+  netSdkVersion: '3.x'
+
+jobs:
+- job: Build
+  pool:
+    vmImage: ${{ parameters.vmImage }}
+
+  variables:
+    BuildConfiguration: 'Release'
+    DOTNET_CLI_TELEMETRY_OPTOUT: 1
+    DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+
+    # We have dependencies on the following Environment Variables:
+    # GITVERSION_PRERELEASETAG
+    # BUILD_REPOSITORY_NAME
+
+    # We have dependencies on the following Build Variables:
+    # Endjin_Service_Connection_NuGet_Org
+    # Endjin_Service_Connection_GitHub
+    # Endjin.ForcePublish
+  steps:
+  - template: install.dotnet-global-tools.workaround.yml
+    parameters:
+      netSdkVersion: ${{ parameters.netSdkVersion }}
+
+  - ${{ parameters.preCustomEnvironmentVariables }}
+
+  - powershell: |
+      Write-Host "##vso[task.setvariable variable=Endjin_IsPreRelease]$((-not ([string]::IsNullOrEmpty($Env:GITVERSION_PRERELEASETAG))))"
+      Write-Host "##vso[task.setvariable variable=Endjin_Repository_Name]$Env:BUILD_REPOSITORY_NAME"
+    displayName: 'Set Environment Variables'
+
+  - powershell: |
+      Write-Host "Initializing $(Build.ArtifactStagingDirectory)/Release"
+      New-Item -Path $(Build.ArtifactStagingDirectory) -Name "Release" -ItemType "directory"
+    displayName: 'Initialize Artifact Staging Release Directory'
+
+  - ${{ parameters.postCustomEnvironmentVariables }}
+
+  # Useful for debugging purposes
+  - powershell: 'gci Env:'
+    condition: or(variables['Endjin.BuildDiagnostics'], variables['Endjin.ShowEnvironment'])
+    displayName: 'Print Environment Variables'
+
+  - ${{ parameters.preBuild }}
+
+  # Executes the 'InvokeBuild' build script
+  - pwsh: |
+      if (!(Get-PSRepository PSGallery)) {
+        Register-PSRepository -Name "PSGallery" `
+                              –SourceLocation "https://www.powershellgallery.com/api/v2/" `
+                              -InstallationPolicy Trusted `
+                              -Force
+      }
+      if (!(Get-Module -ListAvailable InvokeBuild)) {
+        Install-Module InvokeBuild -Repository PSGallery -Force
+      }
+      Import-Module InvokeBuild
+      Invoke-Build "${{ parameters.build_script }}" `
+                   -Task "${{ parameters.build_task }}" `
+                   -Configuration "$(BuildConfiguration)" `
+                   -BuildRepositoryUri "$(Build.Repository.Uri)" `
+                   -SourcesDir "$(Build.SourcesDirectory)" `
+                   -CoverageDir "$(Build.SourcesDirectory)/CodeCoverage" `
+                   -TestReportTypes "HtmlInline_AzurePipelines;Cobertura" `
+                   -PackagesDir "$(Build.ArtifactStagingDirectory)/Packages/$(Build.BuildID)"
+    workingDirectory: $(Build.SourcesDirectory)
+    displayName: Run InvokeBuild script
+
+  - ${{ parameters.preBuild }}
```

Comment:
As I look through the rest of this file I find myself asking: is this exactly the same as the tail end of the normal build? If so, I'm not all that happy about it just being duplicated. Is there a reason that use of `Invoke-Build` can't be made an optional feature of the normal build template? Or alternatively, can we move all the common steps into some new template, and make the "normal" and the `Invoke-Build`-based variants both use that?